pkg/apis/extensions/types.go

```diff
@@ -298,6 +302,47 @@ type DeploymentStatus struct {
 
 	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
 	UpdatedReplicas int `json:"updatedReplicas,omitempty"`
+
+	// Conditions represent the latest available observations of a deployment's current state.
+	Conditions []DeploymentCondition `json:"conditions,omitempty"`
+}
+
+type DeploymentConditionType string
+
+// These are valid conditions of a deployment.
+const (
+	// DeploymentProgressing means the deployment is progressing.
+	DeploymentProgressing DeploymentConditionType = "Progressing"
+	// DeploymentComplete means the deployment has completed progressing.
+	DeploymentComplete DeploymentConditionType = "Complete"
+	// DeploymentFailed means the deployment has failed progressing.
+	DeploymentFailed DeploymentConditionType = "Failed"
+)
+
+// DeploymentCondition describes the state of a deployment at a certain point.
+type DeploymentCondition struct {
+	// Type of deployment condition.
+	Type DeploymentConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status api.ConditionStatus `json:"status"`
+	// LastProbeTime is the last time the condition was checked.
+	LastProbeTime unversioned.Time `json:"lastProbeTime,omitempty"`
+	// LastTransitionTime is the last time the condition transitioned
+	// from one status to another.
+	LastTransitionTime unversioned.Time `json:"lastTransitionTime,omitempty"`
+	// Reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// Message is a human readable message indicating details about
+	// the transition.
+	Message string `json:"message,omitempty"`
+}
+
+// DeploymentRetry is used for retrying a deployment in a failed condition.
+type DeploymentRetry struct {
+	unversioned.TypeMeta `json:",inline"`
+
+	// Name of the deployment to retry.
+	Name string `json:"name"`
```

Comment:
It should be in the subresource path already.

Also, as with the rollback subresource, I think we want annotations, so that the deployment annotations can be updated on retry.